service/history/api/queryworkflow/api.go

```diff
@@ -241,6 +248,27 @@ func Invoke(
 	}
 }
 
+func queryWillTimeoutsBeforeTaskStart(
+	ctx context.Context, shardContext shard.Context, mutableState workflow.MutableState,
+) (bool, error) {
+	startEvent, err := mutableState.GetStartEvent(ctx)
+	if err != nil {
+		return false, err
+	}
+	startAttr := startEvent.GetWorkflowExecutionStartedEventAttributes()
+	workflowTaskBackoffDuration := timestamp.DurationValue(startAttr.GetFirstWorkflowTaskBackoff())
+
+	workflowStart := mutableState.GetExecutionInfo().StartTime.AsTime().UTC()
+	workflowTaskStart := workflowStart.Add(workflowTaskBackoffDuration)
+
+	deadline, _ := ctx.Deadline()
+	deadline = deadline.UTC()
```

Comment:
I think you should check 2nd return value (if deadline is set at all).

Reply:
done in the next rev.